lib/update_grid.rb

```diff
@@ -1,17 +1,16 @@
 # frozen_string_literal: true
 
 class UpdateGrid
-  attr_reader :game_state
-
   def initialize(game_state:)
     @game_state = game_state
     initial_state = [nil, nil, nil, nil, nil, nil, nil, nil, nil]
     @game_state.save(initial_state) if @game_state.state.nil?
   end
 
   def execute(player, position)
-    @game_state.state[position] = player
-    @game_state.save(@game_state.state)
+    @copy = @game_state.state.dup
```

Comment:
what does @copy do?

Reply:
Because we're storing the board in the disk file now rather than RAM, we can't directly update and change it. We create a duplicate of the disk based version of the board which basically creates a version in RAM we can directly update.
We then pass that copy back into save to store it on the disk.